p2p/lightchain.py

```diff
@@ -136,22 +149,90 @@ def callback(r: protocol._DecodedMsgType) -> None:
 
     @alru_cache(maxsize=1024, cache_exceptions=False)
     async def get_account(self, block_hash: Hash32, address: Address) -> Account:
-        peer = cast(LESPeer, self.peer_pool.highest_td_peer)
+        return await self._retry_on_bad_response(
+            partial(self._get_account_from_peer, block_hash, address)
+        )
+
+    async def _get_account_from_peer(
+            self,
+            block_hash: Hash32,
+            address: Address,
+            peer: LESPeer) -> Account:
         key = keccak(address)
         proof = await self._get_proof(peer, block_hash, account_key=b'', key=key)
         header = await self._get_block_header_by_hash(peer, block_hash)
-        rlp_account = HexaryTrie.get_from_proof(header.state_root, key, proof)
+        try:
+            rlp_account = HexaryTrie.get_from_proof(header.state_root, key, proof)
+        except BadTrieProof as exc:
+            raise BadLESResponse("Peer %s returned an invalid proof for account %s at block %s" % (
+                peer,
+                encode_hex(address),
+                encode_hex(block_hash),
+            )) from exc
         return rlp.decode(rlp_account, sedes=Account)
 
     @alru_cache(maxsize=1024, cache_exceptions=False)
-    async def get_contract_code(self, block_hash: Hash32, key: bytes) -> bytes:
-        peer = cast(LESPeer, self.peer_pool.highest_td_peer)
+    @service_timeout(COMPLETION_TIMEOUT)
```

Comment:
Any reason this was not applied to other methods like `get_account`?

Reply:
Not any especially good ones. Added everywhere.